Tests/Unit/MSFT_xDefaultGatewayAddress.Tests.ps1

```diff
@@ -50,193 +80,259 @@ InModuleScope MSFT_xDefaultGatewayAddress {
+                Assert-MockCalled -commandName Get-NetRoute -Exactly 1
+                Assert-MockCalled -commandName Remove-NetRoute -Exactly 1
+                Assert-MockCalled -commandName New-NetRoute -Exactly 1
             }
         }
+    }
 
-        Context 'invoking without -Apply switch and default gateway does not exist' {
+    #######################################################################################
 
-            #region Mocks
-            Mock Get-NetRoute
-            #endregion
+    Describe 'Test-TargetResource' {
+
+        #region Mocks
+        Mock Get-NetAdapter -MockWith { [PSObject]@{ Name = 'Ethernet' } }
+
+        Mock Get-NetRoute -MockWith {
+            [PSCustomObject]@{
+                NextHop = '192.168.0.1'
+                DestinationPrefix = '0.0.0.0/0'
+                InterfaceAlias = 'Ethernet'
+                InterfaceIndex = 1
+                AddressFamily = 'IPv4'
+            }
+        }
+        #endregion
+
+        Context 'checking return with default gateway that matches currently set one' {
+            It 'should return true' {
 
-            It 'when default gateway exists should be $false' {
                 $Splat = @{
                     Address = '192.168.0.1'
                     InterfaceAlias = 'Ethernet'
                     AddressFamily = 'IPv4'
                 }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $false
+                Test-TargetResource @Splat | Should Be $True
             }
+        }
+
+        Context 'checking return with no gateway but one is currently set' {
+            It 'should return false' {
 
-            It 'when default gateway does not exist should be $true' {
                 $Splat = @{
                     InterfaceAlias = 'Ethernet'
                     AddressFamily = 'IPv4'
                 }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $true
-            }
-
-            It 'should call Get-NetRoute twice' {
-                Assert-MockCalled -commandName Get-NetRoute -Exactly 2
+                Test-TargetResource @Splat | Should Be $False
             }
         }
 
-        Context 'invoking with -Apply switch and default gateway exists' {
+        #region Mocks
+        Mock Get-NetRoute -MockWith {}
+        #endregion
 
-            #region Mocks
-            Mock Get-NetRoute -MockWith {
-                [PSCustomObject]@{
-                    NextHop = '192.168.0.1'
-                    DestinationPrefix = '0.0.0.0/0'
+        Context 'checking return with default gateway but none are currently set' {
+            It 'should return false' {
+
+                $Splat = @{
+                    Address = '192.168.0.1'
                     InterfaceAlias = 'Ethernet'
-                    InterfaceIndex = 1
                     AddressFamily = 'IPv4'
                 }
+                Test-TargetResource @Splat | Should Be $False
```

Comment:
This is probably going overboard, but there is ways to capture verbose output, it could be interesting to test that the right verbose streams were used to make sure the right logic was hit. maybe. lol.